### PR TITLE
Nodejs week3

### PR DESCRIPTION
**Pull request by melo4yxa77796**

### CODE REVIEW

File: api/src/routers/meals.js [hunk 1]
```diff
@@ -5,7 +5,80 @@ const mealsRouter = express.Router();
...
+      if (!isNaN(limitValue) && limitValue > 0) {
+        query.limit(limitValue);
+      }
+    }
+
+    if (sortKey !== undefined) {
+      query.orderBy(sortKey);
+    }
+
+    if (sortKey !== undefined) {
```
Comment (kralle333) [inline api/src/routers/meals.js]: sortDir here instead of sortKey? :)

File: api/src/routers/meals.js [hunk 2]
```diff
@@ -5,7 +5,80 @@ const mealsRouter = express.Router();
...
+        return next(error);
+      }
+      const dateString = formattedDate.toISOString().split("T")[0];
+      query.where("created_date", "<", dateString);
+    }
+    if (limit !== undefined) {
+      const limitValue = parseInt(limit, 10);
+      if (!isNaN(limitValue) && limitValue > 0) {
+        query.limit(limitValue);
+      }
```
Comment (kralle333) [inline api/src/routers/meals.js:51]: Should we tell the user if the limit is invalid?

File: api/src/routers/meals.js [hunk 3]
```diff
@@ -5,7 +5,80 @@ const mealsRouter = express.Router();
 
 mealsRouter.get("/", async (req, res, next) => {
   try {
-    const meals = await knex("Meal");
+    const {
```
Comment (kralle333) [inline api/src/routers/meals.js:8]: Great work! You are one of the few of the home works I check where you go through and validate the request and send proper responses. Glad to see it here :)

File: api/src/routers/reviews.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  console.log("GET /api/reviews route called");
+  try {
+    const reviews = await knex("Review");
+    console.log("Reviews retrieved:", reviews);
+    res.json(reviews);
+  } catch (error) {
+    console.error("Error retrieving reviews:", error);
+    next(error);
+  }
+});
```
Comment (kralle333) [inline api/src/routers/reviews.js]: I think you can remove the console.logs here :)

File: api/src/routers/reviews.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    });
+    res.status(201).json({
+      message: "Review created successfully.",
+      reviewId: newReview[0],
+    });
+  } catch (error) {
+    console.error("Error creating review:", error);
+    next(error);
+  }
+});
```
Comment (kralle333) [inline api/src/routers/reviews.js]: good job!

File: api/src/routers/reviews.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+      error:
+        "Fields title, description, meal_id, stars, and created_date are required.",
+    });
+  }
+
+  if (stars < 1 || stars > 5) {
+    return res.status(400).json({
+      error: "Stars must be between 1 and 5.",
+    });
+  }
```
Comment (kralle333) [inline api/src/routers/reviews.js]: this code is duplicated in the post route, can you find a way to not repeat yourself? :)

**Review by kralle333 — APPROVED** [review]
Great job!

You do a great job of validating the requests and giving the proper responses. 
Very very solid work.

Keep it up!

File: api/src/routers/meals.js (hunk 3 above)
Comment (melo4yxa77796) [reply api/src/routers/meals.js:8]: thanks

File: api/src/routers/reviews.js (hunk 4 above)
Comment (melo4yxa77796) [reply api/src/routers/reviews.js]: i did it

File: api/src/routers/reviews.js (hunk 5 above)
Comment (melo4yxa77796) [reply api/src/routers/reviews.js]: thank you

File: api/src/routers/meals.js (hunk 1 above)
Comment (melo4yxa77796) [reply api/src/routers/meals.js]: i made changes

File: api/src/routers/reviews.js (hunk 6 above)
Comment (melo4yxa77796) [reply api/src/routers/reviews.js]: i did it, but i am not sure if it's correct

File: api/src/routers/meals.js (hunk 2 above)
Comment (melo4yxa77796) [reply api/src/routers/meals.js:51]: I think not, but how can I change it?

**Pull request merged**